lib/Wikia/tests/Service/Helios/HeliosClientTest.php

```diff
@@ -49,10 +51,40 @@ public function testSuccess()
 			->method( 'getContent' )
 			->willReturn( '{}' );
 
-		$this->mockStaticMethod( '\Http', 'request', $requestMock );
+		$requestMock->status = $this->getMock( 'stdClass', [ 'isOK', 'getErrorsArray', 'hasMessage' ]);
+		$requestMock->status->expects( $this->any() )->method( 'hasMessage' )->willReturn( false );
+
+		// With no error message, we expect no retries, hence \Http::request should be called only once.
+		$this->getStaticMethodMock( '\Http', 'request' )->expects( $this->once() )
+			->method( 'request' )
+			->willReturn( $requestMock );
 
 		$client = new HeliosClientImpl( 'http://example.com', 'id', 'secret' );
 		$this->assertInternalType( 'object', $client->request( 'resource', [], [], [] ) );
 	}
 
+	public function testRetry()
+	{
+		// To reduce the unit test time, mock the 'sleep' method called between retries.
+		$this->mockGlobalFunction( 'sleep', null);
+
+		$this->mockStaticMethod( '\MWHttpRequest', 'canMakeRequests', true );
+
+		$requestMock = $this->getMock( '\CurlHttpRequest', [ 'execute', 'getContent' ], [ 'http://example.com' ], '', false );
+		$requestMock->expects( $this->once() )
+			->method( 'getContent' )
+			->willReturn( '{}' );
+
+		$requestMock->status = $this->getMock( 'stdClass', [ 'isOK', 'getErrorsArray', 'hasMessage' ]);
```

Comment:
Would it be better to mock `Status` rather than `stdClass` so you're mocking with the expected class that would be stored in that class variable?